synapsis/src/main/java/com/soen/synapsis/appuser/connection/ConnectionService.java

```diff
@@ -1,15 +1,49 @@
 package com.soen.synapsis.appuser.connection;
 
 import com.soen.synapsis.appuser.AppUser;
+import com.soen.synapsis.appuser.AppUserRepository;
 import com.soen.synapsis.appuser.Role;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @Service
 public class ConnectionService {
     private final ConnectionRepository connectionRepository;
+    private final AppUserRepository appUserRepository;
 
-    public ConnectionService(ConnectionRepository connectionRepository) {
+    public ConnectionService(ConnectionRepository connectionRepository, AppUserRepository appUserRepository) {
         this.connectionRepository = connectionRepository;
+        this.appUserRepository = appUserRepository;
+    }
+
+    public List<AppUser> getConnections(Long user_uid) {
+        List<Connection> connectionsRequesters = connectionRepository.findAllByIdRequesterIDAndPendingIsFalse(user_uid);
+        List<Connection> connectionsReceivers = connectionRepository.findAllByIdReceiverIDAndPendingIsFalse(user_uid);
+        List<Long> allConnectionIDs = new ArrayList<>();
+
+        // Get all the receiver IDs
+        if (connectionsRequesters != null) {
+            for (Connection c : connectionsRequesters) {
+                Long receiverID = c.getReceiver().getId();
+                allConnectionIDs.add(receiverID);
+            }
+        }
+        // Get all the requester IDs
+        if (connectionsReceivers != null) {
+            for (Connection c : connectionsReceivers) {
+                Long requesterID = c.getRequester().getId();
+                allConnectionIDs.add(requesterID);
+            }
+        }
+
+        List<AppUser> allConnections = new ArrayList<>();
+        for (Long l : allConnectionIDs) {
+            allConnections.add(appUserRepository.getReferenceById(l));
```

Comment:
This implementation of getting all the connections has the N+1 problem, which should be fine for now as our site is small, but if we notice performance issues here we'll have to modify how this data is retrieved.

Reply:
Hmm, I see.  Just for future reference, how would one fix this issue?